### DISCUSSION TITLE
How Can I add hook to get list of `imported by`.

### DISCUSSION THREAD

**Discussion by byehack** [pyinstaller/pyinstaller#7843] [Help] [answered]
Example `argparse`:
```py
# hooks/before_collecting_modules/hook-argparse.py:

std_mod = ExampleEnum.STD_MODULE
imported_by_list: list

if all(x.mod_type is std_mod for x in imported_by_list):
    excludedimports = ["argparse"]
```
usually stdlib modules lazy import `argparse` in `main()` function. I want to exclude `argparse` when all dependent are from stdlib(or custom list).

is it possible currently?

**Comment by rokm** [comment]
No. Not without hooking every single stdlib module that imports `argparse`.

**Reply by byehack** [reply]
And will you accept those PRs? (many new hooks with `excludedimports = ['argparse']` content)
There is same thing implemented in [hook-pickle.py](https://github.com/pyinstaller/pyinstaller/blob/45fae152e77e9d5267650e05497dabc1a2f882ba/PyInstaller/hooks/hook-pickle.py#L10) with same reason. So, I think you have to accept those PRs 🥲

**Reply by bwoodsend** [reply]
I'd rather people just `--exclude=argparse` if that ~25KB matters so much.

**Reply by byehack** [reply]
What's correct way for `PyInstaller`? then let's make `hook-pickle.py` like others.
Or we can have support needs of proposed single-hook in description.

**Reply by rokm** [reply, marked as answer]
Technically, creating individual hooks like `hook-pickle.py` would be the "correct" approach. 

But I agree with Brénainn - if you are that much concerned about the executable size, use `--exclude=argparse` in your build command to optimize your executable size. If you want to really go hard on size optimization, then except for the most basic programs that use only stdlib, you will probably want to exclude other optional imports from 3rd party packages, anyway...
